app/scripts/minmax/dimMinMax.controller.js

```diff
@@ -169,27 +177,190 @@
 
     angular.extend(vm, {
       active: 'warlock',
+      activesets: '5/5/1',
+      progress: 0,
+      allSetTiers: [],
+      highestsets: {},
+      topsets: [],
+      lockeditems: { helmet: null, gauntlets: null, chest: null, leg: null, classItem: null, ghost: null, artifact: null },
       normalize: 335,
       doNormalize: false,
       type: 'Helmets',
       showBlues: false,
       showExotics: true,
       showYear1: false,
       combinations: null,
+      setOrder: '-str_val,-disc_val,-int_val',
+      setOrderValues: ['-str_val', '-disc_val', '-int_val'],
       statOrder: '-stats.STAT_INTELLECT.value',
       ranked: {},
       filter: {
         int: 3,
         dis: 2,
         str: 2
       },
+      lockedItemsValid: function(dropped_id, dropped_type) {
+          dropped_id = dropped_id.split('-')[1];
+          var item = _.findWhere(buckets[vm.active][dropped_type], {id: dropped_id});
+          var exoticCount = ((item.tier === 'Exotic')? 1 : 0);
+          for(var type in vm.lockeditems) {
+              var item = vm.lockeditems[type];
+              if(item === null || type === dropped_type) { continue; }
+              if(item.tier === 'Exotic' && item.type != 'classItem') {
```

Comment:
I'm fine with any grouping, and sorry for posting the grouping conversation
within the code review type of comments. It's not clear when responding via
email where things will go. Oops.

On Sun, May 22, 2016 at 6:53 PM, kyle shay <notifications@github.com> wrote:

> In app/scripts/minmax/dimMinMax.controller.js
> <https://github.com/DestinyItemManager/DIM/pull/538#discussion_r64161863>:
>
> >        statOrder: '-stats.STAT_INTELLECT.value',
> >        ranked: {},
> >        filter: {
> >          int: 3,
> >          dis: 2,
> >          str: 2
> >        },
> > +      lockedItemsValid: function(dropped_id, dropped_type) {
> > +          dropped_id = dropped_id.split('-')[1];
> > +          var item = _.findWhere(buckets[vm.active][dropped_type], {id: dropped_id});
> > +          var exoticCount = ((item.tier === 'Exotic')? 1 : 0);
> > +          for(var type in vm.lockeditems) {
> > +              var item = vm.lockeditems[type];
> > +              if(item === null || type === dropped_type) { continue; }
> > +              if(item.tier === 'Exotic' && item.type != 'classItem') {
>
> maybe group them? something like:
>
> [image: screen shot 2016-05-22 at 9 50 13 pm]
> <https://cloud.githubusercontent.com/assets/424158/15458384/35e8b6d4-2067-11e6-853d-a7b20040a8cc.png>
>
> or, maybe even better?
>
> https://jsfiddle.net/n47xxpkx/
>
> or maybe something entirely different? open to ideas. -- again this can be
> done in a future iteration.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/DestinyItemManager/DIM/pull/538/files/59aaaec29b25aa17f2eab064b7d5e6b35c64c833#r64161863>
>